### PR TITLE
cks,ui: fix npe and check for disable zone

### LINKED ISSUE #8962: error handling when creating a CKS cluster with no enabled zone
##### ISSUE TYPE

 * Improvement Request

##### COMPONENT NAME

~~~
UI, CKS, Security Groups
~~~

##### CLOUDSTACK VERSION

~~~
4.18
~~~

##### SUMMARY

When a user tries to create a CKS cluster and no zone is enabled a unclear error message is shown. see https://github.com/apache/cloudstack/pull/8489#issuecomment-2069121983

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Fixes #8962

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/version/KubernetesVersionManagerImpl.java [hunk 1]
```diff
@@ -81,6 +81,20 @@ public class KubernetesVersionManagerImpl extends ManagerBase implements Kuberne
 
     public static final String MINIMUN_AUTOSCALER_SUPPORTED_VERSION = "1.15.0";
 
+    protected void updateTemplateDetailsInKubernetesSupportedVersionResponse(
```
Comment (weizhouapache) [inline plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/version/KubernetesVersionManagerImpl.java:84]: @shwstppr 
this looks good
does it fix a known issue ?

Comment (weizhouapache) [reply plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/version/KubernetesVersionManagerImpl.java:84]: great, thanks @shwstppr

**Review by kiranchavala — APPROVED** [review]
LGTM, Able to list the k8s iso if the zone is disabled 

Steps to reproduce the issue


1. Disable a zone
2. Register a k8s iso
3. List k8s ISOs

Before the fix 

```

2024-06-11 07:09:45,983 DEBUG [c.c.a.ApiServlet] (qtp940584193-21:ctx-9fe80c93) (logid:a0bf2315) ===START===  10.0.3.251 -- GET  listall=true&page=1&pagesize=20&command=listKubernetesSupportedVersions&response=json
2024-06-11 07:09:45,983 DEBUG [c.c.a.ApiServlet] (qtp940584193-21:ctx-9fe80c93) (logid:a0bf2315) Two factor authentication is already verified for the user 4, so skipping
2024-06-11 07:09:45,989 DEBUG [c.c.a.ApiServer] (qtp940584193-21:ctx-9fe80c93 ctx-a0cadc52) (logid:a0bf2315) CIDRs from which account 'Account [{"accountName":"ACSUser","id":4,"uuid":"f85ee862-2159-4652-95be-ff08e0d608cb"}]' is allowed to perform API calls: 0.0.0.0/0,::/0
2024-06-11 07:09:45,994 WARN  [o.a.c.a.ProjectRoleBasedApiAccessChecker] (qtp940584193-21:ctx-9fe80c93 ctx-a0cadc52) (logid:a0bf2315) Project is null, ProjectRoleBasedApiAccessChecker only applies to projects, returning API [listKubernetesSupportedVersions] for user [User {"username":"user","uuid":"1a6f567e-0a4f-44fc-af7d-daea5b022d6a"}.] as allowed.
2024-06-11 07:09:45,995 DEBUG [o.a.c.a.StaticRoleBasedAPIAccessChecker] (qtp940584193-21:ctx-9fe80c93 ctx-a0cadc52) (logid:a0bf2315) RoleService is enabled. We will use it instead of StaticRoleBasedAPIAccessChecker.
2024-06-11 07:09:45,995 DEBUG [o.a.c.r.ApiRateLimitServiceImpl] (qtp940584193-21:ctx-9fe80c93 ctx-a0cadc52) (logid:a0bf2315) API rate limiting is disabled. We will not use ApiRateLimitService.
2024-06-11 07:09:45,997 WARN  [c.c.a.d.ParamGenericValidationWorker] (qtp940584193-21:ctx-9fe80c93 ctx-a0cadc52) (logid:a0bf2315) Received unknown parameters for command listKubernetesSupportedVersions. Unknown parameters : listall
2024-06-11 07:09:46,003 ERROR [c.c.a.ApiServer] (qtp940584193-21:ctx-9fe80c93 ctx-a0cadc52) (logid:a0bf2315) unhandled exception executing api command: [Ljava.lang.String;@3677db3c
java.lang.NullPointerException
	at com.cloud.kubernetes.version.KubernetesVersionManagerImpl.createKubernetesSupportedVersionResponse(KubernetesVersionManagerImpl.java:107)
	at com.cloud.kubernetes.version.KubernetesVersionManagerImpl.createKubernetesSupportedVersionListResponse(KubernetesVersionManagerImpl.java:117)
	at com.cloud.kubernetes.version.KubernetesVersionManagerImpl.listKubernetesSupportedVersions(KubernetesVersionManagerImpl.java:278)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)

2024-06-11 07:09:46,006 DEBUG [c.c.a.ApiServlet] (qtp940584193-21:ctx-9fe80c93 ctx-a0cadc52) (logid:a0bf2315) ===END===  10.0.3.251 -- GET  listall=true&page=1&pagesize=20&command=listKubernetesSupportedVersions&response=json
```

After the fix 

```

2024-06-11 08:58:26,612 DEBUG [c.c.a.ApiServlet] (qtp940584193-20:ctx-69fbc1d2) (logid:96c66ed4) ===START===  10.0.3.251 -- GET  listall=true&page=1&pagesize=20&command=listKubernetesSupportedVersions&response=json
2024-06-11 08:58:26,612 DEBUG [c.c.a.ApiServlet] (qtp940584193-20:ctx-69fbc1d2) (logid:96c66ed4) Two factor authentication is already verified for the user 2, so skipping
2024-06-11 08:58:26,618 DEBUG [c.c.a.ApiServer] (qtp940584193-20:ctx-69fbc1d2 ctx-84b74695) (logid:96c66ed4) CIDRs from which account 'Account [{"accountName":"admin","id":2,"uuid":"3948161c-27c8-11ef-af7a-1e00bb0002ca"}]' is allowed to perform API calls: 0.0.0.0/0,::/0
2024-06-11 08:58:26,620 INFO  [o.a.c.a.DynamicRoleBasedAPIAccessChecker] (qtp940584193-20:ctx-69fbc1d2 ctx-84b74695) (logid:96c66ed4) Account [Account [{"accountName":"admin","id":2,"uuid":"3948161c-27c8-11ef-af7a-1e00bb0002ca"}]] is Root Admin or Domain Admin, all APIs are allowed.
2024-06-11 08:58:26,621 WARN  [o.a.c.a.ProjectRoleBasedApiAccessChecker] (qtp940584193-20:ctx-69fbc1d2 ctx-84b74695) (logid:96c66ed4) Project is null, ProjectRoleBasedApiAccessChecker only applies to projects, returning API [listKubernetesSupportedVersions] for user [User {"username":"admin","uuid":"394911fe-27c8-11ef-af7a-1e00bb0002ca"}.] as allowed.
2024-06-11 08:58:26,622 DEBUG [o.a.c.a.StaticRoleBasedAPIAccessChecker] (qtp940584193-20:ctx-69fbc1d2 ctx-84b74695) (logid:96c66ed4) RoleService is enabled. We will use it instead of StaticRoleBasedAPIAccessChecker.
2024-06-11 08:58:26,623 DEBUG [o.a.c.r.ApiRateLimitServiceImpl] (qtp940584193-20:ctx-69fbc1d2 ctx-84b74695) (logid:96c66ed4) API rate limiting is disabled. We will not use ApiRateLimitService.
2024-06-11 08:58:26,626 WARN  [c.c.a.d.ParamGenericValidationWorker] (qtp940584193-20:ctx-69fbc1d2 ctx-84b74695) (logid:96c66ed4) Received unknown parameters for command listKubernetesSupportedVersions. Unknown parameters : listall
2024-06-11 08:58:26,638 DEBUG [c.c.a.ApiServlet] (qtp940584193-20:ctx-69fbc1d2 ctx-84b74695) (logid:96c66ed4) ===END===  10.0.3.251 -- GET  listall=true&page=1&pagesize=20&command=listKubernetesSupportedVersions&response=json

```

**Pull request merged**